src/main/java/com/spotify/docker/client/DockerCertificates.java

```diff
@@ -145,8 +145,26 @@ private PrivateKey readPrivateKey(Path file) throws IOException, InvalidKeySpecE
   private static PrivateKey generatePrivateKey(PrivateKeyInfo privateKeyInfo) throws IOException,
           InvalidKeySpecException, NoSuchAlgorithmException {
     final PKCS8EncodedKeySpec spec = new PKCS8EncodedKeySpec(privateKeyInfo.getEncoded());
-    final KeyFactory kf = KeyFactory.getInstance("RSA");
-    return kf.generatePrivate(spec);
+    return tryGeneratePrivateKey(spec, "RSA", "EC");
```

Comment:
nit: can you put "RSA" and "EC" at the top of the class as a static final ImmutableSet?

like `private static final Set<String> PRIVATE_KEY_ALGS = ImmutableSet.of("RSA", "EC");`

Reply:
Done.